### PR TITLE
metals 0.11.9 (new formula)

### PR DESCRIPTION
**Pull request by henryhchchc**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/metals.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  depends_on "openjdk"
+
+  def install
+    inreplace "build.sbt", /version ~=.+?,/m, "version := \"#{version}\","
+
+    system "sbt", "package"
+
+    # Following Arch AUR to get the dependencies.
+    dep_pattern = /^.+?Attributed\((.+?\.jar)\).*$/
+    sbt_deps_output = `sbt 'show metals/dependencyClasspath' 2>/dev/null`
```
Comment (SMillerDev) [inline Formula/metals.rb:18]:
```suggestion
    sbt_deps_output = Utils.safe_popen_read("sbt 'show metals/dependencyClasspath' 2>/dev/null")
```

Comment (henryhchchc) [reply Formula/metals.rb]: Thanks for the advice!

**Comment by henryhchchc** [comment]
We may convert this PR to draft as I need more time to figure out the `NoClassDefFound` exception on Linux.

**Comment by henryhchchc** [comment]
I think I have fixed all the stuff now.

BTW I leveraged a tricky way to gather the dependencies as the upstream does not provide a way to generate a fat jar. We need to update this formula once fat jar construction is available from upstream.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@henryhchchc, thanks for your contribution to Homebrew! 🎉 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**